### PR TITLE
add rate limiting for LLMs, added a new api for adding RSS , fixed a bug where the RSS interface was not using the access key.

### PR DESCRIPTION
**Pull request by Twelveeee**
# add rate limiting for LLMs, added a new api for adding RSS , fixed a bug where the RSS interface was not using the access key.

## Summary

This PR introduces three major enhancements to improve the functionality, security, and stability of the zenfeed project:

### 1. LLM Rate Limiting Implementation
- Added RPM (Requests Per Minute) configuration field for LLM settings
- Implemented token bucket rate limiter with comprehensive unit tests
- Integrated rate limiting into OpenAI and Gemini LLM implementations
- Updated documentation to describe the new rate limiting configuration

### 2. RSS Feed Source Addition Interface
- Added `AddFeedSource` API endpoint for dynamically adding new RSS feed sources
- Implemented comprehensive validation for feed source parameters
- Added HTTP route `/add_feed_source` for REST API access
- Created MCP tool `add_feed_source` for feed source management through MCP protocol

### 3. Fixed RSS Interface Access Key Bug
- Added `RSSHubAccessKey` configuration field to support RSSHub instances with authentication
- Implemented `appendAccessKeyToURL` method to automatically append access keys to RSSHub API requests
- Updated RSSHub query methods to use access keys properly

---

# 实现LLM 速率限制，新增rss订阅链接添加接口，修复rss接口没有使用accessKey的bug

## 摘要

本 PR 引入了三项主要增强功能，以提高 zenfeed 项目的功能性、安全性和稳定性：

### 1. 实现LLM速率限制
- 为 LLM 设置添加 `RPM`（每分钟请求数）配置字段
- 实现基于令牌桶算法的速率限制器，包含全面的单元测试
- 将速率限制集成到 OpenAI 和 Gemini LLM 实现中
- 更新文档，描述新的速率限制配置

### 2. 新增RSS订阅链接添加接口
- 添加 `AddFeedSource` API 端点，用于动态添加新的 RSS 源
- 实现对 RSS 源参数的全面验证
- 添加 HTTP 路由 `/add_feed_source` 用于 REST API 访问
- 创建 MCP 工具 `add_feed_source`，通过 MCP 协议管理 RSS 源

### 3. 修复RSS接口没有使用accessKey的bug
- 添加 `RSSHubAccessKey` 配置字段，支持需要身份验证的 RSSHub 实例
- 实现 `appendAccessKeyToURL` 方法，自动将访问密钥附加到 RSSHub API 请求
- 更新 RSSHub 查询方法，正确使用访问密钥

## Summary by Sourcery

添加可配置的 LLM 每分钟请求次数限流、新的用于添加 RSS 源的 API 接口（包括 HTTP 和 MCP 工具），并修复 RSSHub 和标签处理逻辑，以提高可靠性并支持带身份认证的 RSS。

New Features:
- 引入可配置的「按 LLM 维度」的每分钟请求数（RPM）限流机制，并将其应用于 OpenAI 和 Gemini provider
- 新增 API endpoint、HTTP route 和 MCP tool，以便动态向抓取配置中添加 RSS 源

Bug Fixes:
- 确保 RSSHub 请求在配置中提供可选的 access key 时，将其追加到所有 RSSHub API URL 中
- 修复 JSON label 的序列化处理，避免返回可能会从对象池中被复用的缓冲区

Enhancements:
- 在 LLM 配置中扩展一个 RPM 字段，并在工厂配置中完成传递
- 在 API 配置中暴露 RSSHub access key，以支持需要认证的 RSSHub 实例

Documentation:
- 在英文和中文的配置指南中补充新的 LLM `rpm` 配置选项说明

Tests:
- 为基于 token-bucket 的 LLM 限流器添加单元测试，覆盖基础行为、速率控制、取消、零/负 RPM，以及并发场景

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Add configurable LLM requests-per-minute rate limiting, a new API surface for adding RSS feed sources (including HTTP and MCP tooling), and fix RSSHub and label handling to improve reliability and authenticated RSS support.

New Features:
- Introduce configurable per-LLM requests-per-minute rate limiting and apply it to OpenAI and Gemini providers
- Add an API endpoint, HTTP route, and MCP tool to dynamically add RSS feed sources to the scrape configuration

Bug Fixes:
- Ensure RSSHub requests include an optional access key from configuration by appending it to all RSSHub API URLs
- Fix JSON label marshaling to avoid returning a buffer that may be reused from the pool

Enhancements:
- Extend LLM configuration with an RPM field and propagate it through factory configuration
- Expose RSSHub access key in API configuration for authenticated RSSHub instances

Documentation:
- Document the new LLM rpm configuration option in both English and Chinese config guides

Tests:
- Add unit tests for the token-bucket-based LLM rate limiter, including basic, rate, cancellation, zero/negative RPM, and concurrency scenarios

</details>

### CODE REVIEW

**Comment by glidea** [comment]
关于客户端限流的‘必要性’，有什么具体场景吗？

File: docs/config-zh.md [hunk 1]
```diff
@@ -43,6 +43,7 @@
 | `llms[].embedding_model` | `string`  | LLM 的 Embedding 模型。例如 `text-embedding-3-small`。如果用于 Embedding，则不能为空。如果此 LLM 被使用，则不能与 `model` 同时为空。**注意：** 初次使用后请勿直接修改，应添加新的 LLM 配置。 |                    | 条件性必需                                     |
 | `llms[].tts_model`       | `string`  | LLM 的文本转语音 (TTS) 模型。                                                                                                                                                                |                    | 否                                             |
 | `llms[].temperature`     | `float32` | LLM 的温度 (0-2)。                                                                                                                                                                           | `0.0`              | 否                                             |
+| `llms[].rpm`             | `int`     | 每分钟请求数限制 (Requests Per Minute)。用于控制 API 调用频率，避免超过提供商的速率限制。`0` 表示不限流。                                          | `0`                | 否                                             |
```
Comment (glidea) [inline docs/config-zh.md:46]: 避免是为了减少报错日志，还有啥

**Comment by Twelveeee** [comment]
> 关于客户端限流的‘必要性’，有什么具体场景吗？

之前用阿里云百炼的 deepseek-v3.2 不到10分钟干完了免费的 一百万token 。

然后我换成用薄荷大佬的公益api，他的RPM只有3，但是免费 (狗头

对于公益大模型来说，RPM限制还是挺有用的，如果一直请求对方服务器，像是ddos对方服务器一样。
或者对于各种免费的 2api 来说，并发太高会有封号风险。
